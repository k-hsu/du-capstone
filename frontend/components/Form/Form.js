import React from "react";
import styled from "@emotion/styled";
import { spacing } from "../../theme";

const StyledForm = styled.form`
  display: flex;
  flex-direction: column;
  gap: ${spacing["1.25"]};
`;

const Form = (props) => <StyledForm {...props} />;

export default Form;
