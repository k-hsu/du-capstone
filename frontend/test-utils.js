import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderHook } from "@testing-library/react-hooks";

function setup(jsx) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}

export * from "@testing-library/react";
export { renderHook, setup };
