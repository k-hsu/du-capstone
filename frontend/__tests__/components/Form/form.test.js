import React from "react";
import { render, screen, renderer } from "../../../test-utils";
import Form from "../../../components/Form/Form";

describe("Form", () => {
  it("should render form component", () => {
    render(<Form>feedback</Form>);
    expect(screen.getByText("feedback")).toBeInTheDocument();
  });
  it("should render form component styles", () => {
    const component = renderer.create(<Form>feedback</Form>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
