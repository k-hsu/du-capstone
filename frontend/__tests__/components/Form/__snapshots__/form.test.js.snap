// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render form component styles 1`] = `
<form
  className="css-1v885rd"
>
  feedback
</form>
`;
